File: 3623. Count Number of Trapezoids I/Solution.java
class Solution {
    public int countTrapezoids(int[][] points) {
        final int mod = (int) 1e9 + 7;
        Map<Integer, Integer> yCount = new HashMap<>();
        for (int[] p : points) yCount.merge(p[1], 1, Integer::sum);
        long total = 0, sqrSum = 0;
        for (long cnt : yCount.values()) {
            long sides = (cnt * (cnt - 1) >> 1);
            total += sides;
            sqrSum += sides * sides;
        }
        long res = (total * total - sqrSum) >> 1;
        return (int) (res % mod);
    }
}

File: 3623. Count Number of Trapezoids I/solution.py
class Solution:
    def countTrapezoids(self, points: List[List[int]]) -> int:
        mod = 10**9 + 7
        total, sqr_sum = 0, 0
        y_cnt = defaultdict(int)
        for p in points:
            v = y_cnt[p[1]]
            y_cnt[p[1]] += 1
            total += v
            sqr_sum += v ** 3
        return ((total * total - sqr_sum) >> 1) % mod